Classes/Hook/ContentPostProc.php

```diff
@@ -117,6 +110,50 @@ public function main(array $params, \TYPO3\CMS\Frontend\Controller\TypoScriptFro
     }
 
     /**
+     * Get all kind of javascript needed in FE
+     *
+     * @return array
+     */
+    protected function getJavascriptForFrontendEditing()
+    {
+        $jsArray = [
+            'userIcon' => $this->iconFactory->getIcon('avatar-default', Icon::SIZE_DEFAULT)->render(),
+            'userName' => $GLOBALS['BE_USER']->user['username'],
+            'loadingIcon' => $this->iconFactory->getIcon('spinner-circle-dark', Icon::SIZE_LARGE)->render(),
+            'iframeUrl' => GeneralUtility::getIndpEnv('TYPO3_SITE_URL') .
+                    'index.php?id=' . $this->typoScriptFrontendController->id .
+                    '&frontend_editing=true',
+            'pageTree' => $this->getPageTreeStructure(),
+            'currentTime' => time(),
+            'labels' => $this->getLocalizedFrontendLabels(),
+        ];
+
+        return $jsArray;
+    }
+
+    /**
+     * Localize labels
+     *
+     * @return array
+     */
+    protected function getLocalizedFrontendLabels()
+    {
+        $languageFactory = GeneralUtility::makeInstance('TYPO3\\CMS\\Core\\Localization\\LocalizationFactory');
+        $parsedLocallang = $languageFactory->getParsedData(
+            'EXT:frontend_editing/Resources/Private/Language/locallang.xlf',
+            'default'
+        );
+        $localizedLabels = [];
+        foreach (array_keys($parsedLocallang['default']) as $key) {
+            if (strpos($key, 'notifications.') === 0 ||
```

Comment:
Yeh, or we need to load all labels into a huge array. Or name all fe-labels 'fe.xxxxx...', or something with typoscript.